src/db-adapters/Mongoose/MongooseAdapter.js

```diff
@@ -614,4 +596,28 @@ export default class MongooseAdapter {
 
     return words.join(" ");
   }
+
+  static getIdQueryType(idOrIds) {
+    const idIsValid = mongoose.Types.ObjectId.isValid;
+    const mode = typeof idOrIds === "string" ? "findOne" : "find";
+    let idQuery;
+
+    if (typeof idOrIds === "string") {
+      if (!idIsValid(idOrIds)) {
+        throw new APIError(404, undefined, "No matching resource found.");
+      }
+
+      idQuery = idOrIds;
+    } else if (Array.isArray(idOrIds)) {
+      let validIds = idOrIds.filter(idIsValid);
+
+      if (!validIds.length) {
+        throw new APIError(404, undefined, "No matching resources found.");
+      }
+
+      idQuery = {"$in": idOrIds};
```

Comment:
This return value has to be wrapped in `{_id: ...}`, or not, but we have to be consistent about it; right now, the `delete` method is assuming the value it gets back won't be wrapped in `{_id:... }`, while `find()` is assuming that it will be wrapped.)